app/models/katello/kt_environment.rb

```diff
@@ -102,6 +100,29 @@ def successor
       self.organization.promotion_paths[0][0] unless self.organization.promotion_paths.empty?
     end
 
+    # creates new env from create_params with self as a prior
+    def insert_successor(create_params, path)
+      new_successor = self.class.create!(create_params)
+      if library?
+        new_successor.prior = self
+        if path
+          old_successor = path.first
+          old_successor.prior = new_successor
+        end
+        new_successor.save
+      elsif successor.nil?
+        new_successor.prior = self
+        new_successor.save
+      else
+        old_successor = successor
+        new_successor.prior = self
+        old_successor.prior = new_successor
+        new_successor.save
+      end
+      fail HttpErrors::UnprocessableEntity, _('An environment is missing a prior') unless all_have_prior?
+      new_successor
+    end
```

Comment:
Two things on this method. First can you refactor out `new_successor.prior = self` and `new_successor.save`? Seems like you duplicate that logic a couple times. Also, I'm wondering if we could/should wrap this method in a transaction.